specs/experimental/zk-fault-dipute-game/overview.md

```diff
@@ -21,6 +34,91 @@
+
+#### Guest Program
+
+For ZK Fault Dispute Game, the guest program is an extension of [L2 Chain Derivation][g-l2-chain-derivation] that
+includes a connectivity check among the blocks from L1 origin block to the specified block `C`. The hash of Block
+`C` is determined as the parent hash stored when the ZK Fault Dispute Game is created by calling `create()` of
+the [Dispute Game Factory].
+
+If the attacker manipulates any data within the extension of L2 Chain Derivation, it will affect the block hash `C`.
+This is because all data is ultimately linked to the block hash `C` through the hash chain. Therefore if any data
+manipulating attack can be thwarted by checking the block hash `C` value. For example, if an attacker creates a proof
+using a transaction that is not included in the [sequencer batch][g-sequencer-batch], the block `C` value will change,
+preventing them from winning the challenge.
+
+[Dispute Game Factory]: https://github.com/ethereum-optimism/specs/blob/46d411bfea922c520a1d43329dbf78a2f6966ae0/specs/fault-proof/stage-one/dispute-game-interface.md#disputegamefactory-interface
+
+#### Host Program
+
+The host program is a main part of the prover, responsible for delivering the guest program and its input to the zkVM.
+The host program first executes the guest program to gather the necessary data, which is same as the data stored in
+[PreImageOracle] during [Optimism's Fault Dispute Game]. For ZKFDG, however, additional data is required, which is the
+L1 block data from the origin of target L2 block to block `C`. Finally, the host program delivers the guest program
+and the preimages to the zkVM to obtain the zkVM proof.
+
+[PreImageOracle]: https://github.com/ethereum-optimism/specs/blob/46d411bfea922c520a1d43329dbf78a2f6966ae0/specs/fault-proof/stage-one/fault-dispute-game.md#preimageoracle
+[Optimism's Fault Dispute Game]: https://github.com/ethereum-optimism/specs/blob/46d411bfea922c520a1d43329dbf78a2f6966ae0/specs/fault-proof/stage-one/fault-dispute-game.md#fault-dispute-game
+
+#### Public Values of Proof
+
+To mark which blocks have been executed, the proof publicly reveals the following data.
+
+``` plain
+1. Output root at the parent block of the target
+2. Output root at the target block
+3. Hash of the block C
+```
+
+#### `ZkVerifier` interface
+
+The proof generated by a prover can be verified on a verifier contract that includes the following interface.
+The verification in a challenge process will be implemented using the verify function provided by the verifier contract.
+
+``` solidity
+interface ZKVerifier {
+    /// @notice The entrypoint for verifying the proof of zk fault proof.
+    /// @param _publicValues The encoded public values.
+    /// @param _proofBytes The encoded proof.
+    /// @return parentOutputRoot The output root at the parent block of the target.
+    /// @return outputRoot The output root at the target block.
+    /// @return L1EndHash The hash of the l1 end block.
```

Comment:
`L1EndHash` seems not to be clear. How about using just `l1BlockHash`? And describe what it is in description.
```suggestion
    /// @return l1BlockHash The block hash of L1 block stored in ZK fault dispute game contract.
```

Reply:
Ok, then how about `l1Head` as used in `FaultDisputeGame`? They used the term `l1Head` for the block hash.

https://github.com/ethereum-optimism/optimism/blob/0cf2a11611d329b9c6d53fd6473dbbc44f68e94c/packages/contracts-bedrock/src/dispute/interfaces/IDisputeGame.sol#L46-L46

Reply:
@dongchangYoo please check this. Excepting this, all looks good to me.

Reply:
It hash been changed.